### PR TITLE
Fix proto package

### PR DESCRIPTION
**Pull request by mconcat**
Closes: #761 

Because of the restriction of protobuf package structure, the gamm proto files(not the pool model files) should be also be under a separate folder, in this PR, `types`. The protobuf does not allow a package with subfolders in it.

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Review by mattverse — COMMENTED** [review]
Oh interesting is it the `error:inconsistent package import paths` or something different that forces us to put all the other files under a sub-directory?

**Review by marbar3778 — COMMENTED** [review]
thank you for doing this!!

you can also go directly to v1 if your api has stabilised

File: proto/osmosis/gamm/v1beta1/types/pool.proto [hunk 1]
```diff
@@ -1,5 +1,5 @@
 syntax = "proto3";
-package osmosis.gamm.v1beta1;
+package osmosis.gamm.v1beta1.types;
```
Comment (ValarDragon) [inline proto/osmosis/gamm/v1beta1/types/pool.proto:2]: Can we undo this? Worried about this breaking things

File: proto/osmosis/gamm/v1beta1/types/query.proto [hunk 2]
```diff
@@ -1,9 +1,9 @@
 syntax = "proto3";
-package osmosis.gamm.v1beta1;
+package osmosis.gamm.v1beta1.types;
```
Comment (ValarDragon) [inline proto/osmosis/gamm/v1beta1/types/query.proto:2]: Can we undo this? Worried about this breaking things

File: proto/osmosis/gamm/v1beta1/types/pool.proto (hunk 1 above)
Comment (marbar3778) [reply proto/osmosis/gamm/v1beta1/types/pool.proto:2]: agree here, having one package for gamm.v1beta1 is better than nesting it in different directories. You have to think that the longer the package name the added few characters add a few bytes to storage.

**Comment by ValarDragon** [comment]
I suspect the types is going to cause @mattverse 's regression test to fail when this is updated to main. (And why we should change the proto package to not include types)

If the regression test doesn't fail, then it doesn't matter!

**Comment by ValarDragon** [comment]
This was really helpful! We learned (the _hard way_) that these proto packages are state breaking. So we changed in #887 to make these all match what was there before identically

**Pull request closed without merging**